catalog/ui/catalog-ui-search/src/main/webapp/js/cql.js

```diff
@@ -72,26 +73,28 @@ define([
         },
 
         follows = {
-            LPAREN: ['NOT', 'GEOMETRY', 'SPATIAL', 'PROPERTY', 'VALUE', 'LPAREN'],
-            RPAREN: ['NOT', 'LOGICAL', 'END', 'RPAREN'],
-            PROPERTY: ['COMPARISON', 'BETWEEN', 'COMMA', 'IS_NULL', 'BEFORE', 'AFTER', 'DURING'],
+            ROOT_NODE: ['NOT', 'GEOMETRY', 'SPATIAL', 'FILTER_FUNCTION', 'PROPERTY', 'LPAREN'],
+            LPAREN: ['NOT', 'GEOMETRY', 'SPATIAL', 'FILTER_FUNCTION', 'PROPERTY', 'VALUE', 'LPAREN', 'RPAREN'],
```

Comment:
Currently never, but we would if start supporting functions that takes 0 arguments. 
http://docs.geoserver.org/stable/en/user/filter/function_reference.html#filter-function-reference

I tried not go overboard with future-proofing but this seemed easy enough to account for.